client/src/hooks/useAuth.tsx

```diff
@@ -0,0 +1,80 @@
+import React, {
+  createContext, useContext, useState, useEffect,
+} from 'react'
+import { IAuthContext, IUser, IAuthContextProps } from '../interfaces'
+import ApiService from '../services/ApiService'
+import JwtService from '../services/JwtService'
+
+const initContext = {
+  user: {
+    id: 0,
+    username: '',
+    password: '',
+    email: '',
+    profileImg: '',
+    headerImg: '',
+  },
+  signIn: () => {},
+  signUp: () => {},
+  signOut: () => {},
+}
+
+export const AuthContext = createContext<IAuthContext>(initContext)
+
+export const useProvideAuth = ():IAuthContext => {
+  const [user, setUser] = useState<IUser | null>(null)
+  const signIn = async (payload:object): Promise<object> => {
+    try {
+      const userInfo = await ApiService.post('/api/v1/users/login', payload)
```

Comment:
i think if we added the `/api/v1` in the base url my be better
```suggestion
      const userInfo = await ApiService.post('/users/login', payload)
```